CryptoAnalysis/src/main/java/crypto/cryslhandler/CrySLModelReader.java

```diff
@@ -118,14 +116,43 @@ public CrySLModelReader() throws MalformedURLException {
 
 	}
 
+	public CrySLRule readRule(InputStream stream, String virtualFileName) {
+		if (!virtualFileName.endsWith(cryslFileEnding))
+			return null;
```

Comment:
So since `null` is not usable for any consumer we should also throw the new exception here. Please pass a message what caused the error (if applicable you can put the exception that caused the error as "inner Exception")